docs/pages/debugging/create-devtools-plugins.mdx

```diff
@@ -0,0 +1,150 @@
+
+The client object returned by `useDevToolsPluginClient` includes:
+<APIBox header="addMessageListener">
+
+Listens for a message matching the type string and invokes the callback with the message data.
+
+```jsx
+const client = useDevToolsPluginClient('my-devtools-plugin');
+client.addMessageListener('ping', (data) => {
+  alert(`Received ping from ${data.from}`);
+});
+```
+
+</APIBox>
+
+<APIBox header="sendMessage">
+
+Listens for a message matching the type string and invokes the callback with the message data.
+
+```jsx
+const client = useDevToolsPluginClient('my-devtools-plugin');
+client?.sendMessage("ping", { from: "web" });
+```
+
+</APIBox>
+
+Edit the Expo app inside the **webui** folder to customize the user interface that displays diagnostic information from your app or triggers test scenarios:
+```tsx webui/App.tsx
+import { useDevToolsPluginClient, type EventSubscription } from 'expo/devtools';
+import { useEffect } from 'react';
+
+export default function App() {
+  const client = useDevToolsPluginClient('my-devtools-plugin');
+
+  useEffect(() => {
+    const subscriptions: EventSubscription[] = [];
+
+    subscriptions.push(
+      client?.addMessageListener('ping', (data) => {
+        alert(`Received ping from ${data.from}`);
+      })
+    );
+
+    return () => {
+      for (const subscription of subscriptions) {
+        subscription?.remove();
+      }
+    };
+  }, [client]);
+}
+
+```
+
+Edit the hook in the **src** folder to customize what diagnostic information is sent to the plugin or how the app should respond to any messages from the web user interface:
+```tsx src/useMyDevToolsPlugin.ts
+import { useDevToolsPluginClient } from 'expo/devtools';
+
+export function useMyDevToolsPlugin() {
+  const client = useDevToolsPluginClient('my-devtools-plugin');
+
+  const sendPing = () => {
+    client?.sendMessage('ping', { from: 'app' });
+  }
+
+  return {
+    sendPing,
+  }
+}
+```
+
+If you update the hook to return functions that will be called by the app, you will also need to update **src/index.ts** so it exports no-op functions when the app is not running in debug mode:
+```diff src/index.ts
+if (process.env.NODE_ENV !== 'production') {
+  useTestPlugin = require('./useTestPlugin').useTestPlugin;
+} else {
+  useTestPlugin = () => ({
++    sendPing: () => {},
+  });
+}
+```
+
+### Test your plugin
+Since the plugin web UI is an Expo app, you can test it just like you would any other Expo app, with `npx expo start`, except that you will run it in the browser only. The template includes a convenience command to run the plugin in local development mode:
+
+<Terminal cmd={['$ npm run web:dev']} />
+
+### Build your plugin
+
+To prepare your plugin for distribution or for use within your monorepo, you will need to build the plugin with the following command:
+
+<Terminal cmd={['$ npm run prepublishOnly']} />
```

Comment:
```suggestion
<Terminal cmd={['$ npm run build:all']} />
```

i changed my mind for the command. even though prepublishOnly still work, but the semantic looks strange. i've added a `build:all` command for this.